### PR TITLE
Update mkosi

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Comment by bluca** [comment]
mkosi centos8 is sad, related?

File: .github/workflows/mkosi.yml [hunk 1]
```diff
@@ -125,7 +125,3 @@ jobs:
 
     - name: Boot ${{ matrix.distro }} QEMU
       run: timeout -k 30 10m mkosi --debug qemu
-
-    # vsock in Github Actions with qemu is broken so for now we check for failures manually.
-    - name: Check ${{ matrix.distro }} QEMU
-      run: sudo mkosi shell bash -c "[[ -e /testok ]] || { cat /failed-services; exit 1; }"
```
Comment (medhefgo) [inline .github/workflows/mkosi.yml:131]: Did you test this by forcing a failure in  `mkosi-check-and-shutdown`?

Comment (daandemeyer) [reply .github/workflows/mkosi.yml:131]: Well I was first going to check whether it works at all now. But I'll add some logic to verify this works as expected

Comment (daandemeyer) [reply .github/workflows/mkosi.yml:131]: We exit with 123 by default now to make sure propagating the exit status works

**Review by bluca — APPROVED** [review]

**Comment by poettering** [comment]
lgtm

**Pull request merged**